packages/flutter_tools/lib/src/build_system/targets/common.dart

```diff
@@ -434,3 +430,10 @@ abstract final class Lipo {
     }
   }
 }
+
+extension EnvironmentExtension on Environment {
```

Comment:
Can `depfileService` just be a final field in `Environment`? extension methods should be avoided per the style guide https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension

Reply:
it could. but then we're adding more "state" to environment.

If that's your preference...